src/enthusiast_custom/examples/pdf_agent/plugins/pdf_documents_plugin.py

```diff
@@ -0,0 +1,32 @@
+from pathlib import Path
+
+import requests
+from enthusiast_common import DocumentSourcePlugin, DocumentDetails
+from langchain_community.document_loaders import PyPDFLoader
+
+
+class PDFDocumentSourcePlugin(DocumentSourcePlugin):
+    def __init__(self, data_set_id: int, config: dict):
+        super().__init__(data_set_id, config)
+
+    def fetch(self) -> list[DocumentDetails]:
+        title = "A practical guide to building agents"
+        url = "https://cdn.openai.com/business-guides-and-resources/a-practical-guide-to-building-agents.pdf"
```

Comment:
This URL (or a set of URLs) could be provided via config - this way we can also show how this concept works